src/main/java/com/kolya/gym/controller/TrainerController.java

```diff
@@ -0,0 +1,141 @@
+package com.kolya.gym.controller;
+
+import com.kolya.gym.data.AuthData;
+import com.kolya.gym.data.TrainerData;
+import com.kolya.gym.domain.Trainer;
+import com.kolya.gym.service.TrainerService;
+import com.kolya.gym.service.UserService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Api(value = "API for trainers", tags = "Trainers")
+@RestController
+@RequestMapping("/trainers")
+public class TrainerController {
+
+    private final TrainerService trainerService;
+    private final UserService userService;
+    private final Logger logger;
+
+
+    @Autowired
+    public TrainerController(TrainerService trainerService, UserService userService, Logger logger) {
+        this.trainerService = trainerService;
+        this.userService = userService;
+        this.logger = logger;
```

Comment:
It's better to instantiate logger within the class itself, rather than providing as injected bean.

Reply:
Fixed